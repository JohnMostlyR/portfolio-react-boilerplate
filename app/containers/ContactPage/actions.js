/*
 *
 * ContactPage actions
 *
 */

import { SEND_FORM, SEND_FORM_SUCCESS, SEND_FORM_ERROR } from './constants';

export function sendForm(field) {
  return {
    type: SEND_FORM,
    status: 'sending',
    field,
  };
}

export function sendFormSuccess() {
  return {
    type: SEND_FORM_SUCCESS,
    status: 'success',
  };
}

export function sendFormError(error) {
  return {
    type: SEND_FORM_ERROR,
    status: 'error',
    error,
  };
}
