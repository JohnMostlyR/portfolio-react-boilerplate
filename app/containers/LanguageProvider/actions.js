/*
 *
 * LanguageProvider actions
 *
 */

import { CHANGE_LANGUAGE } from './constants';

export function changeLanguage(languageLocale) {
  return {
    type: CHANGE_LANGUAGE,
    locale: languageLocale,
  };
}
