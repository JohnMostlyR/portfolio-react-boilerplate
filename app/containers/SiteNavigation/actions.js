/*
 *
 * SiteNavigation actions
 *
 */

import {
  SET_SITE_NAVIGATION_IS_AT_SCREEN_TOP,
  SET_SITE_NAVIGATION_OFFSET_HEIGHT,
  SET_SITE_NAVIGATION_TOP_POSITION,
} from './constants';

export function setSiteNavigationTopPosition(topPosition) {
  return {
    type: SET_SITE_NAVIGATION_TOP_POSITION,
    topPosition,
  };
}

export function setSiteNavigationOffsetHeight(offsetHeight) {
  return {
    type: SET_SITE_NAVIGATION_OFFSET_HEIGHT,
    offsetHeight,
  };
}

export function setSiteNavigationIsAtScreenTop(isAtScreenTop) {
  return {
    type: SET_SITE_NAVIGATION_IS_AT_SCREEN_TOP,
    isAtScreenTop,
  };
}
