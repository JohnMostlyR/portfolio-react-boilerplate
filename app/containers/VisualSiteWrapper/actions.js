/*
 *
 * VisualSiteWrapper actions
 *
 */

import { SET_SCROLL_TOP, SET_SITE_WIDTH } from './constants';

export function setSiteWidth(width) {
  return {
    type: SET_SITE_WIDTH,
    payload: width,
  };
}

export function setScrollTop(topPosition) {
  return {
    type: SET_SCROLL_TOP,
    payload: topPosition,
  };
}
