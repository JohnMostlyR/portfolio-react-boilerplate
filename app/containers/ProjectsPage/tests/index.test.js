import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import createHistory from 'history/createMemoryHistory';

import { ProjectsPage, mapDispatchToProps } from '../index';
import { loadContent } from '../actions';

// enable fake timers
jest.useFakeTimers();

const project = Object.freeze({
  article:
    '# Article for test case\n\n[add attributes to link](https://the.world/)\n',
  description: 'Description for test case',
  images: [
    {
      file: { fileName: 'test-s.png', url: '/test-s.png' },
      title: 'Test image',
    },
    {
      file: { fileName: 'test-m.png', url: '/test-m.png' },
      title: 'Test image',
    },
    {
      file: { fileName: 'test-l.png', url: '/test-l.png' },
      title: 'Test image',
    },
    {
      file: { fileName: 'test-xl.png', url: '/test-xl.png' },
      title: 'Test image',
    },
  ],
  thumbnail: {
    file: {
      contentType: 'image/png',
      details: {
        image: {
          height: 230,
          width: 345,
        },
        size: 22495,
      },
      fileName: 'test-m.png',
      url: '/test-m.png',
    },
    title: 'Test image',
  },
  title: 'Title for test case',
});

/* eslint-disable no-restricted-globals */
describe('<ProjectsPage />', () => {
  const history = createHistory({
    initialEntries: ['/projects'], // The initial URLs in the history stack
    initialIndex: 0, // The starting index in the history stack
    keyLength: 6, // The length of location.key
    getUserConfirmation: null,
  });
  const getContent = jest.fn();
  const componentWillReceivePropsSpy = jest.spyOn(
    ProjectsPage.prototype,
    'componentWillReceiveProps',
  );
  const handleIsLoadingSpy = jest.spyOn(
    ProjectsPage.prototype,
    'handleIsLoading',
  );
  const setIsLoadingSpy = jest.spyOn(ProjectsPage.prototype, 'setIsLoading');

  let wrapper = {};

  afterEach(() => {
    wrapper = null;
    jest.clearAllTimers();
    jest.clearAllMocks();
  });

  it('Expect to render and match the snapshot', () => {
    const { location } = history;
    wrapper = shallow(
      <ProjectsPage
        error={false}
        isLoading={false}
        location={location}
        getContent={getContent}
        projects={[project]}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('fetch content from external source', () => {
    let isLoadingProp = false;

    afterEach(() => {
      wrapper = null;
      jest.clearAllTimers();
      jest.clearAllMocks();
    });

    it('should set the "isLoading" state when the "isLoading" prop is true for more than 200ms', () => {
      wrapper = shallow(
        <ProjectsPage
          error={false}
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[project]}
        />,
      );
      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(0);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);

      isLoadingProp = true;

      wrapper.setProps({ isLoading: isLoadingProp });

      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(1);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().isLoading).toBe(false);

      jest.runTimersToTime(300);

      expect(setIsLoadingSpy).toHaveBeenCalledTimes(1);
      expect(wrapper.state().isLoading).toBe(true);
    });

    it('should not set the "isLoading" state when the "isLoading" prop is true for less than 200ms', () => {
      wrapper = shallow(
        <ProjectsPage
          error={false}
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[project]}
        />,
      );
      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(0);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);

      isLoadingProp = true;

      wrapper.setProps({ isLoading: isLoadingProp });

      expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
      expect(handleIsLoadingSpy).toHaveBeenCalledTimes(1);
      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().isLoading).toBe(false);

      jest.runTimersToTime(100);

      expect(setIsLoadingSpy).toHaveBeenCalledTimes(0);
      expect(wrapper.state().isLoading).toBe(false);
    });
  });

  describe('fetching content from external source', () => {
    afterEach(() => {
      wrapper = null;
      jest.clearAllMocks();
    });

    it('should call getContent() to load external data', () => {
      wrapper = shallow(
        <ProjectsPage
          error={false}
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[project]}
        />,
      );

      expect(getContent).toHaveBeenCalledTimes(1);
    });

    it('should show an error message when content could not be loaded', () => {
      wrapper = shallow(
        <ProjectsPage
          error
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[]}
        />,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should handle case where no content is received', () => {
      wrapper = shallow(
        <ProjectsPage
          error={false}
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[]}
        />,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    describe('getContent', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);
        expect(props.getContent).toBeDefined();
      });
    });

    it('should dispatch loadContent when called', () => {
      const dispatch = jest.fn();
      const props = mapDispatchToProps(dispatch);
      props.getContent();
      expect(dispatch).toHaveBeenCalledWith(loadContent());
    });
  });

  describe('details page', () => {
    afterEach(() => {
      history.go('/projects');
      wrapper = null;
      jest.clearAllTimers();
      jest.clearAllMocks();
    });

    it('Should render the details page when requested', () => {
      history.push('/projects/title-for-test-case');
      const { location } = history;
      wrapper = shallow(
        <ProjectsPage
          error={false}
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[project]}
        />,
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should handle case where a non existent details page is requested', () => {
      history.push('/projects/nothing-here');
      const { location } = history;
      wrapper = shallow(
        <ProjectsPage
          error={false}
          isLoading={false}
          location={location}
          getContent={getContent}
          projects={[project]}
        />,
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
