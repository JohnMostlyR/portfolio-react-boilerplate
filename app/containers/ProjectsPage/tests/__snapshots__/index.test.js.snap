// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsPage /> Expect to render and match the snapshot 1`] = `
<React.Fragment>
  <InjectIntl(HeadGear)
    messages={
      Object {
        "metaDescription": Object {
          "defaultMessage": "This is an overview of my projects. Projects are sorted by date with the most recent on top",
          "id": "portfolio.page.ProjectsPage.meta.description",
        },
        "pageTitle": Object {
          "defaultMessage": "MY PROJECTS",
          "id": "portfolio.page.ProjectsPage.pageTitle",
        },
        "title": Object {
          "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
          "id": "portfolio.page.ProjectsPage.title",
        },
      }
    }
    path="/projects"
  />
  <PageContent
    content={
      <Projects
        projects={
          Array [
            Object {
              "article": "# Article for test case

[add attributes to link](https://the.world/)
",
              "description": "Description for test case",
              "images": Array [
                Object {
                  "file": Object {
                    "fileName": "test-s.png",
                    "url": "/test-s.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-m.png",
                    "url": "/test-m.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-l.png",
                    "url": "/test-l.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-xl.png",
                    "url": "/test-xl.png",
                  },
                  "title": "Test image",
                },
              ],
              "thumbnail": Object {
                "file": Object {
                  "contentType": "image/png",
                  "details": Object {
                    "image": Object {
                      "height": 230,
                      "width": 345,
                    },
                    "size": 22495,
                  },
                  "fileName": "test-m.png",
                  "url": "/test-m.png",
                },
                "title": "Test image",
              },
              "title": "Title for test case",
            },
          ]
        }
        search=""
      />
    }
    noSpeechBubble={true}
    title={
      <FormattedMessage
        defaultMessage="MY PROJECTS"
        id="portfolio.page.ProjectsPage.pageTitle"
        values={Object {}}
      />
    }
  />
</React.Fragment>
`;

exports[`<ProjectsPage /> details page Should handle case where a non existent details page is requested 1`] = `
<React.Fragment>
  <InjectIntl(HeadGear)
    messages={
      Object {
        "metaDescription": Object {
          "defaultMessage": "This is an overview of my projects. Projects are sorted by date with the most recent on top",
          "id": "portfolio.page.ProjectsPage.meta.description",
        },
        "pageTitle": Object {
          "defaultMessage": "MY PROJECTS",
          "id": "portfolio.page.ProjectsPage.pageTitle",
        },
        "title": Object {
          "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
          "id": "portfolio.page.ProjectsPage.title",
        },
      }
    }
    path="/projects/nothing-here"
  />
  <PageContent
    content={
      <Projects
        projects={
          Array [
            Object {
              "article": "# Article for test case

[add attributes to link](https://the.world/)
",
              "description": "Description for test case",
              "images": Array [
                Object {
                  "file": Object {
                    "fileName": "test-s.png",
                    "url": "/test-s.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-m.png",
                    "url": "/test-m.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-l.png",
                    "url": "/test-l.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-xl.png",
                    "url": "/test-xl.png",
                  },
                  "title": "Test image",
                },
              ],
              "thumbnail": Object {
                "file": Object {
                  "contentType": "image/png",
                  "details": Object {
                    "image": Object {
                      "height": 230,
                      "width": 345,
                    },
                    "size": 22495,
                  },
                  "fileName": "test-m.png",
                  "url": "/test-m.png",
                },
                "title": "Test image",
              },
              "title": "Title for test case",
            },
          ]
        }
        search=""
      />
    }
    noSpeechBubble={true}
    title={
      <FormattedMessage
        defaultMessage="MY PROJECTS"
        id="portfolio.page.ProjectsPage.pageTitle"
        values={Object {}}
      />
    }
  />
</React.Fragment>
`;

exports[`<ProjectsPage /> details page Should render the details page when requested 1`] = `
<React.Fragment>
  <InjectIntl(HeadGear)
    messages={
      Object {
        "metaDescription": Object {
          "defaultMessage": "This is an overview of my projects. Projects are sorted by date with the most recent on top",
          "id": "portfolio.page.ProjectsPage.meta.description",
        },
        "pageTitle": Object {
          "defaultMessage": "MY PROJECTS",
          "id": "portfolio.page.ProjectsPage.pageTitle",
        },
        "title": Object {
          "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
          "id": "portfolio.page.ProjectsPage.title",
        },
      }
    }
    path="/projects/title-for-test-case"
  />
  <PageContent
    content={
      <Projects
        projects={
          Array [
            Object {
              "article": "# Article for test case

[add attributes to link](https://the.world/)
",
              "description": "Description for test case",
              "images": Array [
                Object {
                  "file": Object {
                    "fileName": "test-s.png",
                    "url": "/test-s.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-m.png",
                    "url": "/test-m.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-l.png",
                    "url": "/test-l.png",
                  },
                  "title": "Test image",
                },
                Object {
                  "file": Object {
                    "fileName": "test-xl.png",
                    "url": "/test-xl.png",
                  },
                  "title": "Test image",
                },
              ],
              "thumbnail": Object {
                "file": Object {
                  "contentType": "image/png",
                  "details": Object {
                    "image": Object {
                      "height": 230,
                      "width": 345,
                    },
                    "size": 22495,
                  },
                  "fileName": "test-m.png",
                  "url": "/test-m.png",
                },
                "title": "Test image",
              },
              "title": "Title for test case",
            },
          ]
        }
        search=""
      />
    }
    noSpeechBubble={true}
    title={
      <FormattedMessage
        defaultMessage="MY PROJECTS"
        id="portfolio.page.ProjectsPage.pageTitle"
        values={Object {}}
      />
    }
  />
</React.Fragment>
`;

exports[`<ProjectsPage /> fetching content from external source should handle case where no content is received 1`] = `
<React.Fragment>
  <InjectIntl(HeadGear)
    messages={
      Object {
        "metaDescription": Object {
          "defaultMessage": "This is an overview of my projects. Projects are sorted by date with the most recent on top",
          "id": "portfolio.page.ProjectsPage.meta.description",
        },
        "pageTitle": Object {
          "defaultMessage": "MY PROJECTS",
          "id": "portfolio.page.ProjectsPage.pageTitle",
        },
        "title": Object {
          "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
          "id": "portfolio.page.ProjectsPage.title",
        },
      }
    }
    path="blank"
  />
  <React.Fragment />
</React.Fragment>
`;

exports[`<ProjectsPage /> fetching content from external source should show an error message when content could not be loaded 1`] = `
<React.Fragment>
  <InjectIntl(HeadGear)
    messages={
      Object {
        "metaDescription": Object {
          "defaultMessage": "This is an overview of my projects. Projects are sorted by date with the most recent on top",
          "id": "portfolio.page.ProjectsPage.meta.description",
        },
        "pageTitle": Object {
          "defaultMessage": "MY PROJECTS",
          "id": "portfolio.page.ProjectsPage.pageTitle",
        },
        "title": Object {
          "defaultMessage": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
          "id": "portfolio.page.ProjectsPage.title",
        },
      }
    }
    path="blank"
  />
  <ContentLoadingIndicator
    show={true}
    showError={true}
  />
</React.Fragment>
`;
